lib/Alchemy/Phrasea/TaskManager/Job/RecordMoverJob.php

```diff
@@ -252,42 +260,46 @@ private function calcUPDATE(Application $app, $sbas_id, &$sxtask, $playTest)
         }
 
         // compute the 'where' clause
-        list($tw, $join) = $this->calcWhere($app, $sbas_id, $sxtask);
-
-        // ... complete the where to buid the TEST
-        if (count($tws) == 1) {
-            $tw[] = $tws[0];
-        } elseif (count($tws) > 1) {
-            $tw[] = '(' . implode(') OR (', $tws) . ')';
-        }
+        list($tw, $join, $err) = $this->calcWhere($app, $sbas_id, $sxtask);
+
+        if($err == "") {
```

Comment:
What about inverting test. I find it more readable. And else clause becomes useless.

`if (!empty($err)) {
  throw new \Exception($err);
}`